lib/carrierwave/mounter.rb

```diff
@@ -168,7 +168,7 @@ def blank?
 
     def remove=(value)
       @remove = value
-      write_temporary_identifier
+      write_temporary_identifier if remove?
```

Comment:
Doing so is not ideal, as it will prevent users from cancelling the removal by assigning false. (I guess it's better to have a spec for that...)

```ruby
image.remove = true
image.remove = false # this will not call write_temporary_identifier and the attribute change won't be reset
```

I think I found a better solution. Clearing `@original_filename` after store will make `#write_temporary_identifier` harmless, [because `#temporary_identifier` will pick up `@identifier` itself](https://github.com/carrierwaveuploader/carrierwave/blob/96398d9376ef13f47ba4c456b5fd1128e6fcbea8/lib/carrierwave/uploader/proxy.rb#L46). Could you try that approach?

```diff
diff --git a/lib/carrierwave/uploader/store.rb b/lib/carrierwave/uploader/store.rb
index 16a09ca..55d0c06 100644
--- a/lib/carrierwave/uploader/store.rb
+++ b/lib/carrierwave/uploader/store.rb
@@ -87,7 +87,7 @@ module CarrierWave
             end
             @file = new_file
             @identifier = storage.identifier
-            @cache_id = @deduplication_index = nil
+            @original_filename = @cache_id = @deduplication_index = nil
             @staged = false
           end
         end
```

Reply:
Thank you for your review.

I wasn't aware of
> from cancelling the removal by assigning false

this case.

Added some test cases related to remove image cancellation.